webpack/components/ProxmoxContainer/ProxmoxContainerStorage.js

```diff
@@ -0,0 +1,192 @@
+
+const ProxmoxContainerStorage = ({
+  storage,
+  from_profile,
+  storages,
+  nodeId,
+  paramScope,
+}) => {
+  const initData = {
+    id: {
+      name: `${paramScope}[volumes_attributes][0][id]`,
+      value: 'rootfs',
+    },
+    device: {
+      name: `${paramScope}[volumes_attributes][0][device]`,
+      value: '8',
+    },
+    storage: {
+      name: `${paramScope}[volumes_attributes][0][storage]`,
+      value: '',
+    },
+    size: {
+      name: `${paramScope}[volumes_attributes][0][size]`,
+      value: 8,
+    },
+  };
+
+  const [rootfs, setRootfs] = useState(initData);
+
+  useEffect(() => {
+    if (storage && storage.length > 0) {
+      storage.forEach(disk => {
+        if (disk.name === 'rootfs') {
+          setRootfs(disk.value);
+        }
+        if (disk.name === 'mount_point') {
+          addMountPoint(null, disk.value, true);
+        }
+      });
+    }
+  }, [storage]);
+
+  const handleChange = e => {
+    const { name, value } = e.target;
+    const updatedKey = Object.keys(rootfs).find(
+      key => rootfs[key].name === name
+    );
+    const updatedData = {
+      ...rootfs,
+      [updatedKey]: { ...rootfs[updatedKey], value },
+    };
+    setRootfs(updatedData);
+  };
+  const [mountPoints, setMountPoints] = useState([]);
+  const [nextId, setNextId] = useState(1);
+
+  const storagesMap = createStoragesMap(storages, null, nodeId);
+  const addMountPoint = (event, initialData = null) => {
+    if (event) event.preventDefault();
+    const initMP = initialData || {
+      id: {
+        name: `${paramScope}[volumes_attributes][${nextId}][id]`,
+        value: `mp${nextId}`,
+      },
+      device: {
+        name: `${paramScope}[volumes_attributes][${nextId}][device]`,
+        value: `${nextId}`,
+      },
+      storage: {
+        name: `${paramScope}[volumes_attributes][${nextId}][storage]`,
+        value: '',
+      },
+      size: {
+        name: `${paramScope}[volumes_attributes][${nextId}][size]`,
+        value: 8,
+      },
+      volid: {
+        name: `${paramScope}[volumes_attributes][${nextId}][volid]`,
+        value: '',
+      },
+      mp: {
+        name: `${paramScope}[volumes_attributes][${nextId}][mp]`,
+        value: '',
+      },
+    };
+    const newMountPoint = (
+      <MountPoint
+        key={nextId}
+        id={nextId}
+        data={initMP}
+        storagesMap={storagesMap}
+      />
+    );
+    setMountPoints([...mountPoints, newMountPoint]);
+    setNextId(prevId => prevId + 1);
+  };
+
+  const removeMountPoint = idToRemove => {
+    const newMountPoints = mountPoints.filter(
+      mountPoint => mountPoint.props.id !== idToRemove
+    );
+    setMountPoints(newMountPoints);
+  };
+
+  return (
+    <div>
+      <PageSection padding={{ default: 'noPadding' }}>
+        <Title headingLevel="h3">Rootfs</Title>
+        <Divider component="li" style={{ marginBottom: '2rem' }} />
+        <InputField
+          name={rootfs?.storage?.name}
```

Comment:
What happens to the InputField if rootfs or rootfs.storage is not defined?

Reply:
it will be set to nil